regex-syntax/src/parser.rs

```diff
@@ -801,14 +801,25 @@ impl<'a> Chars<'a> {
         }
     }
 
+    /// Skip line comments and whitespace
     fn skip(&mut self) {
         if !self.ignore_space { return; }
         while self.cur < self.chars.len() {
+            // Handle escaping of `#`, i.e. don't start a comment with `\#`.
+            if !self.in_comment && (self.c() == '\\')
+                && ((self.cur + 1) < self.chars.len())
```

Comment:
Could you please use `checkadd(self.cur, 1)` here (and below)? I'm trying to be a bit paranoid about overflow in the parser. (Although I don't have any tests for it...)